### ISSUE TITLE
WebAssembly trapped manipulating ReadableStreams

### ISSUE THREAD

**Issue by roptch** [fastly/js-compute-runtime#21]
Hi,
Following #10, I tried to reproduce a clone response function using `ReadableStream::tee()`, making 2 new responses from one.
The function looks like that:
```js
function cloneResponse(response) {
    const teedOff = response.body.tee();
    return {
        currentResponse: new Response(teedOff[0], { headers: cloneHeaders(response.headers), status: response.status }),
        newResponse: new Response(teedOff[1], { headers: cloneHeaders(response.headers), status: response.status }),
    }
}
```
By doing that, I can read the responses body using the `ReadableStream::getReader()`, they are not locked.
But if I want to use the `Response::json()`, `Response::text()` or `Response::arrayBuffer()` instead of the reader, I get this error:
```
Aug 12 15:57:25.102 ERROR request{id=0}: WebAssembly trapped: wasm trap: unreachable
wasm backtrace:
    0: 0x5a6b - <unknown>!<wasm function 69>
    1: 0x593c - <unknown>!<wasm function 68>
    2: 0x11d6e - <unknown>!<wasm function 181>
    3: 0x129f1 - <unknown>!<wasm function 186>
    4: 0x384ca - <unknown>!<wasm function 475>
    5: 0x316ee - <unknown>!<wasm function 455>
    6: 0x299a9 - <unknown>!<wasm function 452>
    7: 0x387f9 - <unknown>!<wasm function 475>
    8: 0x3b701 - <unknown>!<wasm function 498>
    9: 0x177517 - <unknown>!<wasm function 3190>
   10: 0x96905 - <unknown>!<wasm function 1403>
   11: 0x1236ff - <unknown>!<wasm function 2376>
   12: 0x384ca - <unknown>!<wasm function 475>
   13: 0x3b701 - <unknown>!<wasm function 498>
   14: 0xa5a0e - <unknown>!<wasm function 1557>
   15: 0xfc9a5 - <unknown>!<wasm function 2121>
   16: 0x142cb - <unknown>!<wasm function 219>
   17: 0x3eb380 - <unknown>!<wasm function 8540>
   18: 0x140cf - <unknown>!<wasm function 218>
```
Am I doing something wrong ?
Thanks !

**Comment by tschneidereit** [comment]
Thank you for the report, @roptch!

That is a pretty terrible failure mode, my apologies :(

IIUC, you're teeing the `body`, and then want to use e.g. `Response#text()` on the original response?

(As an aside, you shouldn't need to explicitly clone the headers, that should happen implicitly by just doing `headers: response.headers`. If that doesn't work, that'd be a separate bug.)

**Comment by tschneidereit** [comment]
Ok, that makes sense. I'll look into this and report back!

**Comment by tschneidereit** [comment]
As an update here, I tracked this down to something that's not directly related to `tee()` itself: `Body#text()` et al make the wrong assumption that the `Request` / `Response` operated on is backed by a host request/response, not a content-created `ReadableStream`.

We won't get this fixed before the next release is out, but it shouldn't take too much longer after that.

**Comment by tschneidereit** [comment]
This is fixed by #64—apologies for it taking a while!
